views/AdminView/components/BotNameInput.tsx

```diff
@@ -1,27 +1,42 @@
-import { FC, useCallback, useState } from 'react';
+import { FC, useCallback, useEffect, useState } from 'react';
 import React from 'react-native';
 import {
   Button,
+  CheckBox,
   Input,
   Typography,
+  useCheckBoxValue,
   useInputValue,
 } from '../../../.storybook/stories';
 import { useAppSelector } from '../../../redux/hooks';
 import { BOT_NAME_LENGTH, BOT_SUFFIX } from '../../../utils/env';
-import { BotFather as BotNameInputProps } from '../../../types';
+import { BotFather } from '../../../types';
 import { useSendTelegramMessage } from '../../../hooks';
+import { generateBotName } from '../utils';
+
+type BotNameInputProps = BotFather & { locationName: string };
 
 export const BotNameInput: FC<BotNameInputProps> = ({
+  locationName,
   botFatherAccessHash,
   botFatherId,
 }) => {
   const sendTelegramMessage = useSendTelegramMessage();
   const [isSubmitting, setIsSubmitting] = useState(false);
   const [botUsername, onBotUsernameChange] = useInputValue();
+  const [generateName, onToggleGenerateName] = useCheckBoxValue(true);
```

Comment:
Add `is` prefix to the boolean value.
I guess we should disable input if the value is `true`?